pyscriptjs/src/main.ts

```diff
@@ -61,46 +92,117 @@ class PyScriptApp {
         logger.info('config loaded:\n' + JSON.stringify(this.config, null, 2));
     }
 
-    initialize() {
-        addInitializer(this.loadPackages);
-        addInitializer(this.loadPaths);
-        this.loadRuntimes();
+    // lifecycle (3)
+    showLoader() {
+        // add loader to the page body
+        logger.info('add py-loader');
+        customElements.define('py-loader', PyLoader);
+        this.loader = <PyLoader>document.createElement('py-loader');
+        document.body.append(this.loader);
     }
 
-    loadPackages = async () => {
+    // lifecycle (4)
+    loadRuntime() {
+        logger.info('Initializing runtime');
+        if (this.config.runtimes.length == 0) {
+            showError("Fatal error: config.runtimes is empty");
+            return;
+        }
+
+        if (this.config.runtimes.length > 1) {
+            showError("Multiple runtimes are not supported yet. " +
+                      "Only the first will be used");
+        }
+        const runtime_cfg = this.config.runtimes[0];
+        const runtime: Runtime = new PyodideRuntime(this.config, runtime_cfg.src,
+                                                    runtime_cfg.name, runtime_cfg.lang);
+        this.loader.log(`Downloading ${runtime_cfg.name}...`);
+        const script = document.createElement('script'); // create a script DOM node
+        script.src = runtime.src;
+        script.addEventListener('load', () => {
+            void this.afterRuntimeLoad(runtime);
```

Comment:
Is this just the way you do lambdas in TS?
(Asking because I'm TS noob)

Reply:
yes, it's not even TS-specific, it works the same in plain JS. They are not called `lambda` though, they are just "functions" and/or "closures".
This particular syntax is called "arrow function", which of course has slightly different rules than normal functions :man_facepalming: 

DISCLAIMER: I'm not the author of this code, I just moved it around :)